### PR TITLE
[Docs] `mesheryctl pattern delete` page added

### LINKED ISSUE #2947: [Docs] `mesheryctl pattern delete` page missing
#### Current State
No page exists for this command in the [mesheryctl Command Reference](https://docs.meshery.io/reference/mesheryctl).

#### Desired State
A docs page for this subcommand should be available.

#### Implementation
1. Create an entry for this subcommand in https://github.com/layer5io/meshery/tree/master/docs/_data/mesheryctlcommands/

---
**Contributor Resources**
- [Meshery documentation site](https://docs.meshery.io/)
- [Meshery documentation source](https://github.com/layer5io/meshery/tree/master/docs)
- [Instructions for contributing to documentation](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)

### PR DESCRIPTION
**Pull request by aminoxix**
_Signed-off-by: aminos 🇮🇳  <anshumaankrprasad76@gmail.com>_

**Description**
Try to adding page `mesheryctl pattern delete` for the <a href="https://docs.meshery.io/reference/mesheryctl"> pattern here </a>

This PR fixes #2947 

**Notes for Reviewers**
Ready to merge yet; but changes can be done on reviewer's review
Pl give review for changes, if any....

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by amino19** [comment]
@adithyaakrishna Sir,

Can't able to fix delete as list in `Command` 
Screenshot from 2021-05-30 19-40-42

Pl check this line, I will edit as per requirements:
Screenshot from 2021-05-30 19-43-03

**Comment by adithyaakrishna** [comment]
> @adithyaakrishna Sir,
> 
> Can't able to fix delete as list in `Command`
> Screenshot from 2021-05-30 19-40-42
> 
> Pl check this line, I will edit as per requirements:
> Screenshot from 2021-05-30 19-43-03

@amino19 Can you create a PR for this, will let you know the changes there :)

**Comment by adithyaakrishna** [comment]
@amino19 in Line 318, change the rowspan value from 9 to 10 in the file https://github.com/amino19/meshery/blob/aminos/docs/pages/reference/mesheryctl-commands.md ?

image

**Comment by amino19** [comment]
Yes Done! Perfect.
What about description I made?

File: docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml
Comment (adithyaakrishna) [inline docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml:4]:
```suggestion
    mesheryctl pattern delete --file [path to pattern file]
```

**Review by adithyaakrishna — DISMISSED** [review]

**Comment by amino19** [comment]
I think, I disturb your stale review, sorry for that..
Shall I need to set  `rowspan` as `11`

Facing this :
Screenshot from 2021-05-30 20-45-15

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (adithyaakrishna) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:29]:
```suggestion
  mesheryctl pattern delete --file [path to pattern file] 
```

**Review by adithyaakrishna — CHANGES_REQUESTED** [review]

**Review by alphaX86 — COMMENTED** [review]
Actually we may need to look at refining descriptions.... But it's another task actually, and we're planning it (not now). So for now it's LGTM 👍 

Do changes as @adithyaakrishna requested and we're good to go

**Comment by amino19** [comment]
Done, as suggested by @adithyaakrishna  
Can't we start with all the documents rectification from today?

**Comment by alphaX86** [comment]
> Done, as suggested by @adithyaakrishna
> Can't we start with all the documents rectification from today?

I'm burdened with exams and other stuff till June 3rd. After that only I can able to concentrate. Plus, this can't be done with some persons alone. We need feedback plus, suggestions from Lee, mesheryctl and Docs Contributors for the rectification

**Comment by alphaX86** [comment]
BTW @amino19 update the branch :)

**Comment by amino19** [comment]
Done! 👍🏼

**Comment by alphaX86** [comment]
Now it's good 👍

**Review by alphaX86 — COMMENTED** [review]
LGTM 👍 pinging other reviewers now...

File: docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml
Comment (leecalcote) [inline docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml:2]: @navendu-pottekkat this command doesn't remove the deployment, but simply removes the pattern file, correct?

@utkarsh-pro do we have an endpoint and command plan for deprovisioning infrastructure using the `pattern` command?

Comment (utkarsh-pro) [reply docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml:2]: We don't have another endpoint. We use http method to indicate delete. Basically `POST` on `/api/experimental/pattern/deploy` will provision and `DELETE` on `/api/experimental/pattern/deploy` will deprovision.

File: docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml
Comment (leecalcote) [inline docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml:2]: Can we provide more information to the user? This description isn't helpful beyond what the user already assumes that "delete" does. Can you delete multiple files at-a-time? Does it happen immediately? Can you restore a file? Can you provide a pattern ID or a pattern name? Do you have to provide a file when you're trying to delete a pattern? Why can't you just provide a pattern ID/name instead of having to provide a file, too?

File: docs/pages/reference/mesheryctl-commands.md
Comment (leecalcote) [inline docs/pages/reference/mesheryctl-commands.md:376]: Outside of the need to reference "pattern.delete", why are there changes in these lines? These should all be taken from template.

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (leecalcote) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:3]: Nice title. 👍

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (leecalcote) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:6]: Good use of this option.

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (leecalcote) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:8]: Duplicative language reference. Please consolidate to the appropriate moniker.

File: docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml
Comment (leecalcote) [reply docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml:2]: @utkarsh-pro very good. To clarify, do we have a way for the user to delete only the pattern file and not the infra that it represents?

Comment (utkarsh-pro) [reply docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml:2]: Yes @leecalcote. The above endpoint is only for handling `deployment`. We have separate CRUD endpoints for persisting pattern files. 
`/api/experimental/pattern` `POST GET`
`/api/experimental/pattern/:id` `GET DELETE`

These are the endpoints as far as I remember

**Comment by adithyaakrishna** [comment]
@amino19 Can you make the changes @leecalcote suggested and Except those everything else LGTM 🚀💯

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (adithyaakrishna) [reply docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:6]: @leecalcote I think this exists in all the files present in the master branch, https://github.com/layer5io/meshery/blob/master/docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-apply.md
This would also be taken up during the complete overhauling of meshery docs 👀

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md
Comment (amino19) [reply docs/pages/reference/mesheryctl/subcommands/mesheryctl-pattern-delete.md:3]: Thanks Sir! :)

File: docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml
Comment (amino19) [reply docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml]: I made it as `delete pattern file will simply remove the pattern file using specific flags`
Is that correct now?

Comment (amino19) [reply docs/_data/mesheryctlcommands/pattern/delete/commands/delete.yaml]: Needs your approval to go!

File: docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml
Comment (amino19) [reply docs/_data/mesheryctlcommands/pattern/delete/flags/delete-file.yaml]: Screenshot from 2021-06-01 14-16-13

Thanks @utkarsh-pro sir, for helping me!
This takes a lengthy format, so I will make it like:
Screenshot from 2021-06-01 14-33-17

File: docs/pages/reference/mesheryctl-commands.md
Comment (amino19) [reply docs/pages/reference/mesheryctl-commands.md:376]: @leecalcote Sir, I think this exist in every steps!

**Comment by amino19** [comment]
@leecalcote : Browse the preview: https://deploy-preview-3070--meshery-docs.netlify.app

**Comment by leecalcote** [comment]
Thanks. Even more helpful would be this link - https://deploy-preview-3070--meshery-docs.netlify.app/reference/mesheryctl/pattern/delete

**Comment by leecalcote** [comment]
The lack of punctuation in these sentences and all CAPS “delete” needs to be fixed.

End users do not need to know which API endpoint is invoked by a mesheryctl command. 

Users don’t know what “the persisters” means and neither do I.

**Comment by amino19** [comment]
@leecalcote : Done with changes!

**Review by navendu-pottekkat — COMMENTED** [review]
Thank you @amino19 

The sentences should start with a Capital letter. The rest looks good.

Screenshot from 2021-06-03 00-39-43

**Comment by amino19** [comment]
@navendu-pottekkat : Sir, Changes had been made as per instruction!
PR : ready to merge :rocket:

**Review by navendu-pottekkat — APPROVED** [review]

**Comment by navendu-pottekkat** [comment]
Thank you @amino19 🚀

**Comment by leecalcote** [comment]
Excellent. :)

**Pull request merged**
